### PR TITLE
Changed sizing to be automatic based on ID/Type

### PR DESCRIPTION
**Pull request by jesse-deboer**
## PR Summary

**Changes**
- Changed to automatic sizing instead of having to set it yourself
- Removed Infortis compatability

**Introduced a helper class for image dimensions**
- Added a new helper class to obtain image dimensions from the view configuration.
- Revised image size options
- Removed small and large options and added a default option for all images.

**Modified plugin to use custom URL with width/height parameters**
- The plugin now uses custom URLs with width/height parameters based on the current image size in Magento 2's view configuration for product images.

### CODE REVIEW

File: Model/Image.php [hunk 1]
```diff
@@ -22,20 +22,11 @@ public function __construct(
         $this->storeManager = $storeManager;
     }
 
-    public function getDefaultUrl(string $currentUrl): string
+    public function getCustomUrl(string $imageUrl, int $width, int $height): string
     {
-        return $this->getServiceUrl($currentUrl, $this->config->getConfigValue(Config::XPATH_FIELD_LARGE));
+        return $this->getServiceUrl($imageUrl, $this->config->getConfigValue(Config::XPATH_FIELD_DEFAULT));
```
Comment (JeroenBoersma) [inline Model/Image.php]: width and height are unused, is this correct?

File: Model/Image.php [hunk 2]
```diff
@@ -22,20 +22,11 @@ public function __construct(
         $this->storeManager = $storeManager;
     }
 
-    public function getDefaultUrl(string $currentUrl): string
```
Comment (JeroenBoersma) [inline Model/Image.php:25]: you could be backwards compatible when you leave this method `getDefaultUrl` in the model

File: Model/Image.php [hunk 3]
```diff
@@ -22,20 +22,11 @@ public function __construct(
...
+    public function getCustomUrl(string $imageUrl, int $width, int $height): string
     {
-        return $this->getServiceUrl($currentUrl, $this->config->getConfigValue(Config::XPATH_FIELD_LARGE));
+        return $this->getServiceUrl($imageUrl, $this->config->getConfigValue(Config::XPATH_FIELD_DEFAULT));
     }
 
-    public function getSmallUrl(string $currentUrl): string
```
Comment (JeroenBoersma) [inline Model/Image.php:30]: same goes for these, you could maybe make these backwards compatible

**Review by JeroenBoersma — CHANGES_REQUESTED** [review]
have added some feedback, awesome MR thought!

File: Model/Image.php (hunk 1 above)
Comment (jesse-deboer) [reply Model/Image.php]: Fixed!

File: Model/Image.php (hunk 3 above)
Comment (jesse-deboer) [reply Model/Image.php:30]: Due to everything now using the custom url this has been deleted. All values are now generated with a width and height and these settings are removed.

File: Model/Image.php [hunk 4]
```diff
@@ -22,19 +22,9 @@ public function __construct(
...
-
-    public function getSmallUrl(string $currentUrl): string
-    {
-        return $this->getServiceUrl($currentUrl, $this->config->getConfigValue(Config::XPATH_FIELD_SMALL));
-    }
-
-    public function getAutoCompleteUrl(string $currentUrl): string
-    {
-        return $this->getServiceUrl($currentUrl, $this->config->getConfigValue(Config::XPATH_FIELD_AUTOCOMPLETE));
+        return $this->getServiceUrl($imageUrl, 'w=' . $dimensions['width'] . '&h=' . $dimensions['height'] . '&auto=compress&auto=format');
```
Comment (JeroenBoersma) [inline Model/Image.php]: you don't need `$dimensions[]` here, go with `$width` and `$height`

Comment (JeroenBoersma) [reply Model/Image.php]: The possibility to have some extra options `&auto=compress&auto=format` after for instance, maybe something to reconsider

**Review by JeroenBoersma — CHANGES_REQUESTED** [review]

File: Model/Image.php (hunk 4 above)
Comment (jesse-deboer) [reply Model/Image.php]: Made a whoopsie there! Will add other options like cropping with selector in the admin later

**Pull request merged**